src/system_reader.rs

```diff
@@ -0,0 +1,927 @@
+            }
+        };
+
+        Ok(Some(SystemStreamItem::SymbolTableData(ion_type, is_null)))
+    }
+
+    // Called when the system reader advances to the `imports` field of an LST.
+    fn move_to_lst_imports_field(&mut self, ion_type: IonType) -> IonResult<()> {
+        match ion_type {
+            IonType::Symbol => {
+                // If the `imports` field value is the symbol '$ion_symbol_table', then this is an
+                // LST append.
+                self.load_current_symbol()?;
+                if self
+                    .lst
+                    .current_symbol
+                    .matches(system_symbol_ids::ION_SYMBOL_TABLE, "$ion_symbol_table")
+                {
+                    self.lst.is_append = true;
+                }
+            }
+            IonType::List => {
+                // Once this is supported, this branch will do nothing because the list either
+                // be processed when the user steps into/through it or when they try to skip over
+                // it, not when it's first encountered. For now though, we fail because this
+                // feature is not yet supported.
+                todo!("Support shared symbol table imports.");
+            }
+            _ => {
+                // Non-list, non-symbol values for the `imports` field are ignored.
+            }
+        };
+        Ok(())
+    }
+
+    // Reads the raw reader's current value expecting a symbol. Stores the value in
+    // `self.lst.current_symbol` so it can be returned if the user requests it.
+    fn load_current_symbol(&mut self) -> IonResult<()> {
+        let token = self
+            .raw_reader
+            .read_symbol()?
+            .expect("load_current_symbol() called at a value that was not a symbol.");
+        self.lst.current_symbol = token;
+        Ok(())
+    }
+
+    // Reads the raw reader's current value expecting a string. Stores the value in
+    // `self.lst.current_string` so it can be returned if the user requests it.
+    fn load_current_string(&mut self) -> IonResult<()> {
+        self.lst.current_string.clear();
+        let SystemReader {
+            ref mut raw_reader,
+            ref mut lst,
+            ..
+        } = *self;
+        raw_reader
+            .string_ref_map(|s| lst.current_string.push_str(s))
+            .expect("load_current_string() called at a value that was not a string.");
+        Ok(())
+    }
+
+    // Reads the raw reader's current value expecting an integer. Stores the value in
+    // `self.lst.current_int` so it can be returned if the user requests it.
+    fn load_current_int(&mut self) -> IonResult<()> {
+        // Note: This method will only be called on integers found inside of local symbol tables.
+        //       If an LST has an integer that's too big to fit in an i64, this will fail.
+        self.raw_reader
+            .read_i64()?
+            .expect("load_current_int() called at a value that was not an integer.");
```

Comment:
Could someone who writes an LST with some really large integer value could cause the library to panic?  If so, this should be mapped to some error state.